core/kernel/tee_ta_manager.c

```diff
@@ -843,6 +843,139 @@ TEE_Result tee_ta_invoke_command(TEE_ErrorOrigin *err,
 	return res;
 }
 
+#if defined(CFG_TA_STATS)
+static TEE_Result tee_ta_dump_memstats(struct tee_ta_session *s,
+				       uint32_t cancel_req_to,
+				       struct tee_ta_param *param)
+{
+	TEE_Result res = TEE_SUCCESS;
+	struct tee_ta_ctx *ctx = NULL;
+	struct ts_ctx *ts_ctx = NULL;
+
+	ts_ctx = s->ts_sess.ctx;
+	if (ts_ctx)
+		ctx = ts_to_ta_ctx(ts_ctx);
```

Comment:
tee_ta_dump_memstats() is called from `tee_ctxes`/`open_sessions` iteration under tee_ta_mutex lock.
if there is no session, it will not enter this func and call `ts_ctx->ops->dump_mem_stats(&s->ts_sess);`.